lib/question_answer/comment_creator.rb

```diff
@@ -8,12 +8,26 @@ def self.create(attributes)
       ActiveRecord::Base.transaction do
         if qa_comment.save
           create_commented_notification(qa_comment)
+
+          DB.after_commit do
+            publish_changes(qa_comment)
+          end
         end
       end
 
       qa_comment
     end
 
+    def self.publish_changes(qa_comment)
+      Scheduler::Defer.later "Publish new Q&A comment" do
```

Comment:
Just wondering why this needs to be deferred?

Reply:
I did this for performance reasons. Publishing a message is not urgent so I don't want to add the overhead to the request.

Reply:
Ah okay, no problem, I guess a lot of the time these type of publishes are already within jobs etc. so I just haven't noticed before.